File: mobile/native/arucon-health/ios/AruconHealth/AruconHealthModule.swift
import ExpoModulesCore

/**
 Build-linked boundary only. This module intentionally imports no HealthKit API,
 asks for no permission, and performs no health query.
 */
public class AruconHealthModule: Module {
  public func definition() -> ModuleDefinition {
    Name("AruconHealth")

    AsyncFunction("getContractAsync") { () -> [String: Any] in
      return [
        "contractVersion": 1,
        "readMode": "disabled",
        "platform": "ios"
      ]
    }

    AsyncFunction("inspectAvailabilityAsync") { (_: String) -> [String: String] in
      return ["status": "unavailable", "reason": "native_read_disabled"]
    }

    AsyncFunction("getReadPermissionAsync") { (_: String) -> String in
      return "not_requested"
    }
  }
}

File: mobile/native/arucon-health/android/src/main/java/com/arucon/health/AruconHealthModule.kt
package com.arucon.health

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

/**
 * Build-linked boundary only. There is deliberately no Health Connect client,
 * permission launcher, record read, or background worker in this module.
 */
class AruconHealthModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("AruconHealth")

    AsyncFunction("getContractAsync") {
      mapOf(
        "contractVersion" to 1,
        "readMode" to "disabled",
        "platform" to "android"
      )
    }

    AsyncFunction("inspectAvailabilityAsync") { _: String ->
      mapOf("status" to "unavailable", "reason" to "native_read_disabled")
    }

    AsyncFunction("getReadPermissionAsync") { _: String ->
      "not_requested"
    }
  }
}
